canvas-discussion-bot/index.js

```diff
@@ -0,0 +1,137 @@
+const axios = require('axios');
+const dotenv = require('dotenv');
+const { faker } = require('@faker-js/faker');
+const fs = require('fs');
+const path = require('path');
+const puppeteer = require('puppeteer');
+
+async function retrieveCourses(apiToken, canvasURL) {
+  try {
+    const endPoint = "/api/v1/accounts/self/courses";
+    const url = canvasURL + endPoint;
+    
+    const response = await axios.get(url, {
+      headers: {
+        'Authorization': `Bearer ${apiToken}`
+      }
+    });
+    
+    return response.data;
+  } catch (error) {
+    throw new Error(`Error retrieving courses: ${error.message}`);
+  }
+}
+
+// Specify the path to the .env file for the local directory
+const envFilePath = path.resolve(__dirname, '.env');
+
+// Load the .env file
+const result = dotenv.config({ path: envFilePath });
+
+if (result.error) {
+  console.error('Error loading .env file:', result.error);
+} else {
+  console.log('Successfully loaded .env file');
+}
+
+async function postDiscussion(username, password, courseId, index) {
+  const browser = await puppeteer.launch({ headless: false });
```

Comment:
Is there a reason this can't be run in headless mode?

Reply:
Yes, it can. Again, that was just so I could see what was happening.